src/Integrations/Integrations/PDO/PDOIntegration.php

```diff
@@ -133,6 +133,18 @@ public static function detectError($pdoOrStatement, SpanData $span)
         }
         $errorInfo = $pdoOrStatement->errorInfo();
         $span->meta[Tag::ERROR_MSG] = 'SQL error: ' . $errorCode . '. Driver error: ' . $errorInfo[1];
+
+        // Driver-specific error message will be in the rest of the array
+        // The spec suggests the error message to be at pos 2 only,
+        // but e.g. SQLServer doc states that the array can contain monre than 3 elements
+        if (count($errorInfo) > 2) {
+            $span->meta[Tag::ERROR_MSG] .= '. Driver-specific error message: ' ;
+
+            for ($i = 2; $i < count($errorInfo); $i++) {
+                $span->meta[Tag::ERROR_MSG] .= $errorInfo[$i];
+            }
```

Comment:
I would break at the first non-string argument and use the `implode` function to put separators only when appropriate.

```suggestion
            $additionalInfo = [];
            for ($i = 2; $i < count($errorInfo); $i++) {
                if (! is_string($errorInfo[$i])) {
                    break;
                }
                $additionalInfo[] = $errorInfo[$i];
            }
            $span->meta[Tag::ERROR_MSG] .= implode('. ', $additionalInfo);
```